cli/pcluster/cfnconfig.py

```diff
@@ -372,136 +163,168 @@ def __init__(self, args):  # noqa: C901 FIXME!!!
                         )
                     else:
-                        self.__cluster_template = __config.get("global", "cluster_template")
+                        cluster_template = self.__config.get("global", "cluster_template")
             except AttributeError:
-                self.__cluster_template = __config.get("global", "cluster_template")
-        self.__cluster_section = "cluster %s" % self.__cluster_template
-        self.parameters["CLITemplate"] = self.__cluster_template
+                cluster_template = self.__config.get("global", "cluster_template")
 
-        # Check if package updates should be checked
+        return cluster_template
+
+    def __check_for_updates(self):
+        """Check for updates, if required."""
+        # verify if package updates should be checked
         try:
-            self.__update_check = __config.getboolean("global", "update_check")
+            update_check = self.__config.getboolean("global", "update_check")
         except configparser.NoOptionError:
-            self.__update_check = True
+            update_check = True
 
-        if self.__update_check is True:
+        if update_check is True:
             try:
-                __latest = json.loads(
+                latest = json.loads(
                     urllib.request.urlopen("http://pypi.python.org/pypi/aws-parallelcluster/json").read()
                 )["info"]["version"]
-                if self.version < __latest:
-                    print("warning: There is a newer version %s of AWS ParallelCluster available." % __latest)
+                if self.version < latest:
+                    print("warning: There is a newer version %s of AWS ParallelCluster available." % latest)
             except Exception:
                 pass
 
-        # Check if config sanity should be run
+    def __init_sanity_check(self):
+        """
+        Check if config sanity should be run and initialize the corresponding attribute.
+
+        The method also initializes the ResourceValidator object, to be used to validate the resources.
+        """
         try:
-            self.__sanity_check = __config.getboolean("global", "sanity_check")
+            self.__sanity_check = self.__config.getboolean("global", "sanity_check")
+            self.__resource_validator = ResourceValidator(
+                self.region, self.aws_access_key_id, self.aws_secret_access_key
+            )
         except configparser.NoOptionError:
             self.__sanity_check = False
+
         # Only check config on calls that mutate it
-        __args_func = self.args.func.__name__
-        if (
-            __args_func == "create" or __args_func == "update" or __args_func == "configure"
-        ) and self.__sanity_check is True:
+        args_func = self.args.func.__name__
+        if (args_func == "create" or args_func == "update" or args_func == "configure") and self.__sanity_check is True:
```

Comment:
`if args_func in ["create", "update", "configure"]`

Reply:
Fixed thank you for the suggestion.